File: FRONT/AmusementPark/src/app/models/images/image-dto.ts
import { ImageCategory } from './image-category';
import { ImageOwnerType } from './image-owner-type';
import { ImageGeoLocation } from './image-geo-location';
import { LocalizedItemDto } from '../shared/localized-item-dto';

export interface ImageDto {
  id: string;
  category: ImageCategory;
  ownerType: ImageOwnerType;
  ownerId?: string;
  path?: string;
  description?: string;
  isCurrent: boolean;
  isPublished: boolean;
  isWatermarked: boolean;
  width: number;
  height: number;
  sizeInBytes: number;
  originalFileName?: string;
  contentType?: string;
  sourceUrl?: string | null;
  geoLocation?: ImageGeoLocation | null;
  exifMetadata?: ImageExifMetadataDto | null;
  altTexts: LocalizedItemDto<string>[];
  captions: LocalizedItemDto<string>[];
  credits: LocalizedItemDto<string>[];
  tagIds: string[];
  createdAt: string;
  updatedAt: string;
}

export interface ImageExifMetadataDto {
  cameraMaker?: string | null;
  cameraModel?: string | null;
  takenOnUtc?: string | null;
  orientation?: string | null;
  focalLength?: number | null;
  aperture?: number | null;
  exposureTime?: number | null;
  iso?: number | null;
}

File: API/Dtos/Dtos/Images/ImageDto.cs
using Dtos.Images.Creating;
using Dtos.Shared;

namespace Dtos.Images
{
    public sealed class ImageDto
    {
        public required string Id { get; set; }
        public required ImageCategoryDto Category { get; set; }
        public required ImageOwnerTypeDto OwnerType { get; set; }
        public string? OwnerId { get; set; }
        public string? Path { get; set; }
        public string? Description { get; set; }
        public bool IsCurrent { get; set; }
        public bool IsPublished { get; set; }
        public int Width { get; set; }
        public int Height { get; set; }
        public long SizeInBytes { get; set; }
        public string? OriginalFileName { get; set; }
        public string? ContentType { get; set; }
        public ImageGeoLocationDto? GeoLocation { get; set; }
        public List<LocalizedItemDto<string>> AltTexts { get; set; } = new();
        public List<LocalizedItemDto<string>> Captions { get; set; } = new();
        public List<LocalizedItemDto<string>> Credits { get; set; } = new();
        public List<string> TagIds { get; set; } = new();
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }
}
